src/hex_tarball.erl

```diff
@@ -317,12 +321,11 @@ try_updating_mtime(Path) ->
     _ = file:write_file_info(Path, #file_info{mtime=Time}, [{time, universal}]),
     ok.
 
-create_tarball(Files, Options) ->
-    Tarball = create_memory_tarball(Files),
-    case proplists:get_bool(compressed, Options) of
-        true -> gzip(Tarball);
-        false -> Tarball
-    end.
+build_tarball(Files) ->
+    create_memory_tarball(Files).
+
+compress_tarball(Tarball) ->
+    gzip(Tarball).
```

Comment:
build_tarball/1 and compress_tarball/1 seem like unnecessary indirections, let's call these functions directly.